### PR TITLE
Refactor- Fiber, FiberCache and OapIndexType

### PR DESCRIPTION
**Pull request by gczsjdy**
## What changes were proposed in this pull request?

- Add `toFiberCache` function
- Add `FiberType`
- `AnyIndexType` -> `OapIndexType`
- Extract `Fiber`, `FiberCache` to new files

## How was this patch tested?
Existing tests.

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/MemoryManager.scala [hunk 1]
```diff
@@ -233,30 +85,30 @@ private[oap] object MemoryManager extends Logging {
...
       bytes,
       Platform.BYTE_ARRAY_OFFSET,
       memoryBlock.getBaseObject,
       memoryBlock.getBaseOffset,
       bytes.length)
-    DataFiberCache(memoryBlock)
+    if (fiberType == FiberType.Index) {
+      IndexFiberCache(memoryBlock)
+    } else {
+      DataFiberCache(memoryBlock)
```
Comment (linhong-intel) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/MemoryManager.scala]: Actually, `IndexFiberCache` and `DataFiberCache` are identical.
Shall we merge them into one class?

Comment (gczsjdy) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/MemoryManager.scala]: Ok, then I suppose `FiberType` is not needed?

**Comment by linhong-intel** [comment]
LGTM

**Comment by gczsjdy** [comment]
Thanks @linhong-intel

**Pull request merged**